lib/kube/proxy/testing/kube_server/crd.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeserver
+
+import (
+	"net/http"
+	"path/filepath"
+	"sort"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var teleportRoleList = metav1.List{
+	TypeMeta: metav1.TypeMeta{
+		Kind:       "TeleportRoleList",
+		APIVersion: "resources.teleport.dev/v6",
+	},
+	ListMeta: metav1.ListMeta{
+		ResourceVersion: "1231415",
+	},
+	Items: []runtime.RawExtension{
+		{
+			Object: newTeleportRole("telerole-1", "default"),
+		},
+		{
+			Object: newTeleportRole("telerole-1", "default"),
+		},
+		{
+			Object: newTeleportRole("telerole-2", "default"),
+		},
+		{
+			Object: newTeleportRole("telerole-test", "default"),
+		},
+		{
+			Object: newTeleportRole("telerole-1", "dev"),
+		},
+		{
+			Object: newTeleportRole("telerole-2", "dev"),
+		},
+	},
+}
+
+func newTeleportRole(name, namespace string) *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetKind("TeleportRole")
+	obj.SetAPIVersion("resources.teleport.dev/v6")
+	obj.SetName(name)
+	obj.SetNamespace(namespace)
+	return obj
+}
+
+func (s *KubeMockServer) listTeleportRoles(w http.ResponseWriter, req *http.Request, p httprouter.Params) (any, error) {
```

Comment:
I don't love that these have a return value but also accept a responsewriter - makes it hard to tell what you're supposed to do in order to write a response body. 

Was this just necessary because you had to stick with the same signature we use in teleport's web API?